geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientRegistrationEventQueueManager.java

```diff
@@ -143,7 +154,7 @@ public ClientRegistrationEventQueue create(
     final ClientRegistrationEventQueue clientRegistrationEventQueue =
         new ClientRegistrationEventQueue(eventQueue,
             putDrainLock);
-    registeringProxyEventQueues.put(clientProxyMembershipID,
+    registeringProxyEventQueues.putIfAbsent(clientProxyMembershipID,
```

Comment:
and this will avoid accidental overwrite by the second/losing client

Reply:
Correct - since we can have multiple threads registering the same client proxy membership ID, we only want one queue between them, so whoever gets their first will create the queue.